File: pkg/runtime/javascript/fixtures/update/can_update.airplane.js
import airplane from "airplane";

const obj = {}

export const t1 = airplane.task(
  {
    ...obj,
    slug: "spread",
  },
  async () => {
    return []
  },
);

// This task does not have a slug, so we can't add a test for it. However, its
// existence helps test whether the `good` task below can be updated when it exists
// in a file with a task like this.
export const t2 = airplane.task(
  obj,
  async () => {
    return []
  },
);

export const t3 = airplane.task(
  {
    slug: "computed",
    parameters: process.env.AIRPLANE_ENV_SLUG === "prod" ? {
      confirm: "boolean"
    } : {}
  },
  async () => {
    return []
  },
);

const name = "parameters"
export const t4 = airplane.task(
  {
    [name]: {},
    slug: "key",
  },
  async () => {
    return []
  },
);

export const t5 = airplane.task(
  {
    slug: "template",
    name: `Computed ${name}`
  },
  async () => {
    return []
  },
);

export const t6 = airplane.task(
  {
    slug: "tagged_template",
    name: sql`Computed ${name}`
  },
  async () => {
    return []
  },
);

// An example task that can be updated that is contained within a file
// with tasks that cannot be updated.
export const good = airplane.task(
  {
    slug: "good",
    // Various value types that should not be flagged as computed:
    a: undefined,
    b: null,
    c: 123,
    d: 12.34,
    e: "hello",
    f: 'world',
    // The following two template literals are not computed.
    g: `test`,
    h: sql`test`,
    i: ["hello"],
    j: {jj: "world"},
    k: true,
    l: false,
  },
  async () => {
    return []
  },
);

File: pkg/definitions/updaters/python/fixtures/can_update_airplane.py
import airplane
import os

from mylib import MyGlobalDefaultDateTime


# This is not valid and should not get matched on.
@airplane.task()
class MyTask:
    @airplane.task()
    def run(self):
        pass


@airplane.task(slug="deploy" if os.getenv("AIRPLANE_ENV") == "prod" else "deploy_dev")
def computed_slug():
    pass


@airplane.task(restrict_callers=["task"] if os.getenv("AIRPLANE_ENV") == "prod" else [])
def conditional():
    pass

restrict_callers = ["task"]

@airplane.task(restrict_callers=restrict_callers)
def shared_value():
    pass

MY_DEFAULT_CONSTANT = "foo"

@airplane.task()
def shared_default(val: str = MY_DEFAULT_CONSTANT):
    pass


@airplane.task(name=f"My task ({os.getenv('AIRPLANE_ENV')})")
def template():
    pass


@airplane.task(
    # This template string has no computed expressions, so it's fine.
    name=f"My task",
    # Various value types that should not be flagged as computed:
    a=None,
    c=123,
    d=12.34,
    e="hello",
    f="world",
    i=["hello"],
    j={"jj": "world"},
    k=True,
    l=False,
)
def good():
    pass
